pkg/build/api/v1/types.go

```diff
@@ -303,6 +303,9 @@ type BuildStrategy struct {
 
 	// CustomStrategy holds the parameters to the Custom build strategy
 	CustomStrategy *CustomBuildStrategy `json:"customStrategy,omitempty" description:"holds parameters to the Custom build strategy"`
+
+	// ExternalStrategy holds the parameters to the External build strategy
+	ExternalStrategy *ExternalBuildStrategy `json:"externalStrategy,omitempty" description:"holds parameters to the External build strategy"`
```

Comment:
Think of the build config as representing an entire build process.  The external engine will write back a pretty detailed status blob that will either be strictly API versioned (generic steps, tasks, stages, pipelines, whatever) or we'll figure out how to let it evolve and still have naive UI clients show it.  If that external process uses builds, other build configs, deployment configs, or things not on the platform, they'll set the appropriate labels / annotations to let naive clients do something kind of useful, but mostly drive status.